quinn-h3/benches/bench.rs

```diff
@@ -106,77 +119,41 @@ impl Context {
-        debug!("server bind");
+    pub fn spawn_server(&mut self) -> (SocketAddr, thread::JoinHandle<()>) {
+        let socket = UdpSocket::bind(SocketAddr::new(IpAddr::V6(Ipv6Addr::LOCALHOST), 0)).unwrap();
+        let addr = socket.local_addr().unwrap();
+        let server = self.server_config.clone();
+
+        let (stop_server, stop_recv) = oneshot::channel::<()>();
+        self.stop_server = Some(stop_server);
+
         let handle = thread::spawn(move || {
-            let my_span = span!(Level::TRACE, "server");
-            let _enter = my_span.enter();
             let mut runtime = rt();
-            let handle = runtime.spawn(
-                async move {
-                    let mut incoming = server.build().unwrap();
-                    let mut incoming_req = incoming
-                        .next()
-                        .await
-                        .expect("accept")
-                        .await
-                        .expect("connect");
-                    debug!("recv incoming");
-                    while let Some(recv_req) = incoming_req.next().await {
-                        let (request, mut body_reader, sender) = recv_req.await.expect("recv_req");
-                        while let Some(_) = body_reader.data().await {}
-
-                        let mut body_writer = sender
-                            .send_response(
-                                Response::builder().status(StatusCode::OK).body(()).unwrap(),
-                            )
-                            .await
-                            .expect("send_response");
-
-                        let frame_size = request
-                            .headers()
-                            .get("frame_size")
-                            .map(|x| x.to_str().unwrap().parse().expect("parse frame size"))
-                            .expect("no frame size");
-                        let mut remaining = request
-                            .headers()
-                            .get("total_size")
-                            .map(|x| x.to_str().unwrap().parse().expect("parse total size"))
-                            .expect("no total size");
-
-                        let data = "a".repeat(frame_size);
-                        while remaining > 0 {
-                            let size = std::cmp::min(frame_size, remaining);
-                            body_writer
-                                .write_all(&data.as_bytes()[..size])
-                                .await
-                                .expect("body write");
-                            remaining -= size;
-                        }
-                    }
-                }
-                .instrument(error_span!("server")),
-            );
+            let handle = runtime.spawn(async {
+                let incoming_conn = server.with_socket(socket).unwrap();
+                handle_connection(incoming_conn, stop_recv).instrument(error_span!("server"))
```

Comment:
Missing `.await`. This is why the benchmark is timing out instead of running.

Reply:
Thanks! Knew it was obvious. I must have missed a warning then :/